qucs/qucs/qucs.cpp

```diff
@@ -2284,6 +2284,10 @@ void QucsApp::slotSelectSubcircuit(const QModelIndex &idx)
 
   QString filename = idx.sibling(idx.row(), 0).data().toString();
   QString note = idx.sibling(idx.row(), 1).data().toString();
+  int idx_pag = DocumentTab->currentIndex();
+  QString tab_titl = "";
+  if (idx_pag>=0) tab_titl = DocumentTab->tabText(idx_pag);
+  if (filename == tab_titl ) return; // Forbid to paste subcircuit into itself.
```

Comment:
I have not looked so much into hashes. But there is something related to hash function in:
https://github.com/Qucs/qucs/blob/master/qucs/qucs/components/subcircuit.cpp
Line: 331

Doesn't that uniquely identify a schematic?